packages/widget/src/components/amount-selector/amount-selector.ts

```diff
@@ -45,26 +48,31 @@ export class AmountSelector extends BaseComponent {
 
   tokenBalanceController = new TokenBalanceController(this);
 
+  _handleOnResourceSelectedCallback = (): void => {
```

Comment:
Maybe better to move that into existing resourceSelected callback: https://github.com/sygmaprotocol/sygma-widget/pull/137/files#diff-27d2b4e4c285a859d1c78d56d9586464abcf08c45067c54bbb0d17a322fddfb2R89 ?

Reply:
Also why is this change in this PR?

Reply:
i think the logic is extracted because it's used in 2 places (i usually do this when i see it's used >=3 times) 🤷

Reply:
Exactly, I created a method as I used it twice. The problem with the max button is when we set the value this.amount the "@change"  _onInputAmountChangeHandler will not be triggered to call this.onResourceSelected inside the method. So I call it manually inside the _useMaxBalance